libnmstate/netapplier.py

```diff
@@ -149,6 +149,7 @@ def _apply_ifaces_state(
                 )
             if not nmclient.can_disable_ipv6():
                 _disable_ipv6(desired_state)
+            nmclient.delete_client()
```

Comment:
We may have a resource leak here in case of errors (exceptions raised which causes this line not to be reached).
This is true in all places where you use this resource release function.

I think there is a need for an nmclient contextmanager with a try/finally structure.
If we wrap it with a decorator, we can decorate the libnmstate public functions (apply, commit, rollback, show) in a clean way without worrying about the inner details... keeping us safe if other changes are done inside the functions/flow.

This is also true for the nm integration tests, we need there a fixture that does it per test/module (with `autouse=True`?)